### PR TITLE
Drop Rails < 6.0, Ruby < 2.7, test with Ruby 3.1, 3.2, and 3.3.

### PR DESCRIPTION
**Pull request by HeyNonster**
Rails 5.2 and Ruby 2.6 went end-of-life in 2018 so I think they're both safe to drop support for. This PR also starts testing with more recent Ruby versions.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -72,85 +64,13 @@ workflows:
...
-                gemfile: gemfiles/rails5.1_graphql1.9.gemfile
-              - ruby-version: '3.0'
-                gemfile: gemfiles/rails5.2_graphql1.8.gemfile
-              - ruby-version: '3.0'
-                gemfile: gemfiles/rails5.2_graphql1.9.gemfile
-              - ruby-version: '3.0'
-                gemfile: gemfiles/rails5.2_graphql1.10.gemfile
-              - ruby-version: '3.0'
-                gemfile: gemfiles/rails5.2_graphql1.11.gemfile
+              exclude:
```
Comment (mcelicalderon) [inline .circleci/config.yml:67]: @HeyNonster indentation is 2 spaces too much to the right here, that's why CI is failing to run as the matrix is production way too many jobs

Comment (HeyNonster) [reply .circleci/config.yml]: Should be fixed now!

File: .circleci/config.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
       RUBYOPT: '-rostruct'
     steps:
       - checkout
-      - run: gem install bundler -v '1.17'
+      - run: gem install bundler -v '2.5.23'
```
Comment (HeyNonster) [inline .circleci/config.yml:21]: Bumping this in the Ruby 3.1+ commit because we seemed to be getting failures without it.

**Comment by 00dav00** [comment]
I also think we should remove them :muscle:

**Comment by HeyNonster** [comment]
@mcelicalderon and @00dav00 how would you like to handle GraphQL < 1.12 support? It seems like GraphQL `1.11` has some issues with later Ruby versions. We can either drop support for `graphql-ruby < 1.12` or skip testing with Ruby 3.1+ and GraphQL 1.11.

**Comment by HeyNonster** [comment]
@mcelicalderon and @00dav00 how would you like to handle GraphQL < 1.12 support? It seems like GraphQL `1.11` has some issues with later Ruby versions. We can either drop support for `graphql-ruby < 1.12` or skip testing with Ruby 3.1+ and GraphQL 1.11.

> I see, @HeyNonster. I think we can just skip those failing combinations in the matrix for now. In the future and before another version release, we might drop support for GQL older versions. That might happen when we get to #273

Hey! I've been at Ruby Conf this week so I haven't been able to finish this but I'll pick it back up next week!

**Comment by mcelicalderon** [comment]
@HeyNonster @00dav00 after reviewing this one, I think I'm going to merge it as is. It will break master, but it's just the build and it would probably fail right now too as it fails to resolve dependencies. I'm working on https://github.com/graphql-devise/graphql_devise/pull/277 and it will be easier to do with this changes and I will fix the remaining jobs.

Thanks again, @HeyNonster!

**Pull request merged**